homeassistant/components/linkplay/media_player.py

```diff
@@ -0,0 +1,240 @@
+"""Support for LinkPlay media players."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from linkplay.bridge import LinkPlayBridge, LinkPlayMultiroom
+from linkplay.consts import LoopMode, PlayingStatus
+from linkplay.controller import LinkPlayController
+from linkplay.exceptions import LinkPlayException
+
+from homeassistant.components import media_source
+from homeassistant.components.media_player import (
+    BrowseMedia,
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaType,
+    RepeatMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    BRIDGE_DISCOVERED,
+    CONTROLLER,
+    DEFAULT_FEATURES,
+    DOMAIN,
+    EQUALIZER_MAP,
+    EQUALIZER_MAP_INV,
+    REPEAT_MAP,
+    REPEAT_MAP_INV,
+    SEEKABLE_FEATURES,
+    SOURCE_MAP,
+    SOURCE_MAP_INV,
+    STATE_MAP,
+)
+from .utils import get_active_multiroom, get_info_from_project
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up a media player from a config entry."""
+
+    @callback
+    def add_entity(bridge: LinkPlayBridge) -> None:
+        async_add_entities([LinkPlayMediaPlayerEntity(bridge)])
+
+    entry.async_on_unload(async_dispatcher_connect(hass, BRIDGE_DISCOVERED, add_entity))
+
+
+class LinkPlayMediaPlayerEntity(MediaPlayerEntity):
+    """Representation of a LinkPlay media player."""
+
+    _bridge: LinkPlayBridge
+
+    def __init__(self, bridge: LinkPlayBridge) -> None:
+        """Initialize the LinkPlay media player."""
+        self._bridge = bridge
+
+        self._attr_unique_id = self.entity_id = f"{DOMAIN}.{self._bridge.device.uuid}"
+        self._attr_name = self._bridge.device.name
+        self._attr_sound_mode_list = list(EQUALIZER_MAP.values())
+        self._attr_should_poll = True
+        self._attr_device_class = MediaPlayerDeviceClass.RECEIVER
+        self._attr_source_list = [
+            SOURCE_MAP[playing_mode]
+            for playing_mode in self._bridge.device.playmode_support
+        ]
+        self._attr_media_content_type = MediaType.MUSIC
+
+    @property
+    def device_info(self) -> dr.DeviceInfo:
+        """Return the device info."""
+        manufacturer, model = get_info_from_project(
+            self._bridge.device.properties["project"]
+        )
+        return dr.DeviceInfo(
+            configuration_url=self._bridge.endpoint,
```

Comment:
This is not a web UI for users to use. So should not be offered.
```suggestion
```

Reply:
I don't know if it is specific to the media player I'm using, but when you browse to the endpoint you have several options to control the media player and its settings. 

![image](https://github.com/home-assistant/core/assets/26272906/3a77b117-0c40-4299-954b-3ecabf69db1b)